extensions/smallrye-metrics/deployment/src/main/java/io/quarkus/smallrye/metrics/deployment/SmallRyeMetricsProcessor.java

```diff
@@ -141,4 +142,44 @@ public void logCleanup(BuildProducer<LogCleanupFilterBuildItem> logCleanupFilter
         logCleanupFilter.produce(new LogCleanupFilterBuildItem("io.smallrye.metrics.MetricsRegistryImpl",
                 "Register metric ["));
     }
+
+    @BuildStep
+    @Record(STATIC_INIT)
+    void registerMetricsFromAnnotatedMethods(SmallRyeMetricsRecorder metrics,
+            BeanContainerBuildItem beanContainerBuildItem,
+            BeanArchiveIndexBuildItem beanArchiveIndex) {
+        IndexView index = beanArchiveIndex.getIndex();
+        JandexBeanInfoAdapter beanInfoAdapter = new JandexBeanInfoAdapter(index);
+        JandexMemberInfoAdapter memberInfoAdapter = new JandexMemberInfoAdapter(index);
+
+        for (DotName metricAnnotation : METRICS_ANNOTATIONS) {
+            Collection<AnnotationInstance> metricAnnotationInstances = index.getAnnotations(metricAnnotation);
+            for (AnnotationInstance metricAnnotationInstance : metricAnnotationInstances) {
+                AnnotationTarget metricAnnotationTarget = metricAnnotationInstance.target();
+                switch (metricAnnotationTarget.kind()) {
+                    case CLASS: {
+                        ClassInfo clazz = metricAnnotationTarget.asClass();
+                        BeanInfo beanInfo = beanInfoAdapter.convert(clazz);
+                        for (MethodInfo method : clazz.methods()) {
+                            metrics.registerMetrics(beanInfo, memberInfoAdapter.convert(method));
+                        }
+                        break;
+                    }
+                    case METHOD: {
```

Comment:
Does this result in redefining metrics if there multiple different Metric annotations on the same method or class?

Reply:
I am not sure I follow.

By `..result in redefining..` you mean `..result in the metric being registered more than once`?